src/main/java/com/google/sps/data/potentialMatchAlgorithm.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.Set;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.data.friend_map.UserFriendsMap;
+import com.google.sps.data.friend_map.UserNode;
+
+/**
+* Class that contains methods that find the potential matches for users.
+*
+* <p>Potential matches are users with who a particular user shares at least one friend with.
+*/
+public class potentialMatchAlgorithm {
```

Comment:
nit: should the class name be capitalized?

Reply:
Yes, it should be! I'll update that. I also ended up doing this for the test file, so I'll update that too.